task.c

```diff
@@ -0,0 +1,120 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "task.h"
+#include "common.h"
+#include "sockets.h"
+
+/* Tasks are stored in a consecutive chunk of memory, the first
+ * sizeof(task_spec) bytes are arranged acording to the struct
+ * task_spec. Then there is an array of task_args of length
+ * (num_args + num_returns), and then follows the data of
+ * pass-by-value arguments of size args_value_size. The offsets in the
+ * task_arg.val are with respect to the end of the struct, i.e.
+ * with respect to the address &task_spec.ids[0]. */
+
+typedef struct {
+  /* Either ARG_BY_REF or ARG_BY_VAL. */
+  int8_t type;
+  union {
+    unique_id arg_id;
```

Comment:
In Ray it is going to be an object id, but it could be any abstract identifier that identifies an argument, like a "delayed" in dask. Not a super big deal and if you prefer obj_id, I'm happy to rename it ;)